src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/CollectionsShouldImplementGenericInterface.cs

```diff
@@ -96,18 +98,94 @@ public override void Initialize(AnalysisContext analysisContext)
                 return;
             }
 
-            var allInterfaces = namedTypeSymbol.AllInterfaces.Select(t => t.OriginalDefinition).ToImmutableHashSet();
+            var allInterfaces = namedTypeSymbol.AllInterfaces.Select(t => t.OriginalDefinition).ToImmutableArray();
```

Comment:
:question: This changes the behavior from before, specifically in the case of types that implement both `IEnumerable<A>` and `IEnumerable<B>`. Was this change intentional? (I believe it does not change behavior but I still wanted to ask.)

:bulb: Consider using `SelectAsArray` for efficiency. Even better, simply `foreach` over `AllIterfaces` or the result of `Select` directly .

Reply:
The change was not intentional, but I agree that it should not make a difference in implementation, except in cases where there are lots of implementations of any of the generic interfaces, but I don't think that will make much a difference here.